packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -71,17 +71,23 @@ const checkIfShouldShowSurvey = (settings) => {
   const { enabled, lastResponseDate, firstDismissalDate, lastDismissalDate } = settings;
 
   // This function goes through all the cases where we'd want to not show the survey:
-  // 1. If the survey is disabled, abort mission, don't bother checking the other settings.
-  // 2. If the user has already responded to the survey, check if enough time has passed since the last response.
-  // 3. If the user has dismissed the survey twice or more before, check if enough time has passed since the last dismissal.
-  // 4. If the user has only dismissed the survey once before, check if enough time has passed since the first dismissal.
+  // 1. If the survey is disabled by strapi, abort mission, don't bother checking the other settings.
+  // 2. If the survey is disabled by user, abort mission, don't bother checking the other settings.
+  // 3. If the user has already responded to the survey, check if enough time has passed since the last response.
+  // 4. If the user has dismissed the survey twice or more before, check if enough time has passed since the last dismissal.
+  // 5. If the user has only dismissed the survey once before, check if enough time has passed since the first dismissal.
   // If none of these cases check out, then we show the survey.
   // Note that submitting a response resets the dismissal counts.
-  // Checks 3 and 4 should not be reversed, since the first dismissal will also exist if the user has dismissed the survey twice or more before.
+  // Checks 4 and 5 should not be reversed, since the first dismissal will also exist if the user has dismissed the survey twice or more before.
 
   // For users who had created an account before the NPS feature was introduced,
   // we assume that they would have enabled the NPS feature if they had the chance.
 
+  // Global strapi disable for NSP.
+  if (window?.strapi?.flags === undefined || window.strapi.flags.nps === false) {
```

Comment:
maybe dumb question, but what circumstance do you see `window.strapi.flags` being false? 😄 the only one i can think of would be SSR envs but we don't support that.

Reply:
For some reason in the fronted test it was complaining about flags being false

Reply:
ah yes, you'll want to update this file: https://github.com/strapi/strapi/blob/0608d433f0396c305dadbe7300c1c9715869b34b/packages/admin-test-utils/src/environment.ts#L59

Reply:
Thanks I missed that.